File: btypes_primitives/src/main/js/de/hhu/stups/btypes/BString.ts
import {BObject} from "./BObject";
import {BBoolean} from "./BBoolean";

export class BString implements BObject {
	value: string;

	constructor(value: string) {
		this.value = value;
	}

	getValue() {
		return this.value;
	}

	equals(o): boolean {
		if (this == o)
			return true;
		if (o == null || o !instanceof BString)
			return false;

		return this.value === o.value;
	}

	length(): number {
		return this.value.length;
	}

	isEmpty(): boolean {
		return this.value === "";
	}

	toString(): string {
		return '"' + this.value + '"';
	}

	isCase(o): boolean {
		return this.value === o;
	}

	isString(): BBoolean {
		return new BBoolean(true);
	}

	isNotString(): BBoolean {
		return new BBoolean(false);
	}
}

File: btypes_big_integer/src/main/java/de/hhu/stups/btypes/BString.java
package de.hhu.stups.btypes;

import java.util.Objects;

public final class BString implements BObject {

	private final String value;

	public BString(String value) {
		this.value = Objects.requireNonNull(value, "value");
	}

	public String getValue() {
		return this.value;
	}

	@Override
	public boolean equals(Object o) {
		if (this == o) {
			return true;
		} else if (!(o instanceof BString)) {
			return false;
		} else {
			return this.value.equals(((BString) o).value);
		}
	}

	@Override
	public int hashCode() {
		return this.value.hashCode();
	}

	@Override
	public String toString() {
		return '"' + this.value + '"';
	}

	public BBoolean isString() {
		return BBoolean.TRUE;
	}

	public BBoolean isNotString() {
		return BBoolean.FALSE;
	}

	public BBoolean equal(BString o) {
		return BBoolean.of(this.equals(o));
	}

	public BBoolean unequal(BString o) {
		return BBoolean.of(!this.equals(o));
	}
}
